### PR TITLE
Fix building bspline C extension on Windows

### PR DESCRIPTION
**Pull request by finagle29**
The bspline C extension currently fails to build on Windows (e.g. see the win_64 build in conda-forge/staged-recipes/#12629/checks). This PR adds a ``PyInit__bspline`` function to ``bpsline.c`` (necessary because the ``distutils`` windows compiler expects a ``PyInit_MODULE_NAME`` symbol), as well as adding ``export_symbols`` to ``setup_package.py``.

### CODE REVIEW

**Review by profxj — APPROVED** [review]
Looks fine to me, but I don't know anything about Windows.  ;)

**Review by kbwestfall — APPROVED** [review]
Thanks for doing this!  Are there any Windows-specific installation recommendations/requirements that we should add to the install docs?

**Comment by finagle29** [comment]
Getting a weird error when running the unit tests on Windows now. I'll dig into it, but just wanted to note that this PR is not ready to merge yet.

win_error2

Kyle, I don't think installation on Windows was that different from my experience of installation on macOS.

**Comment by finagle29** [comment]
Unit tests pass on Windows and macOS!
`============================= 246 passed, 676 warnings in 1692.66s (0:28:12) =============================` on Windows
`================ 246 passed, 292 warnings in 929.48s (0:15:29) =================` on macOS

File: pypeit/bitmask.py [hunk 1]
```diff
@@ -351,12 +351,12 @@ def turn_on(self, value, flag):
 
         out = value | (1 << self.bits[_flag[0]])
         if len(_flag) == 1:
-            return out
+            return out.astype(value.dtype)
```
Comment (profxj) [inline pypeit/bitmask.py:354]: Worth @kbwestfall checking this.

**Comment by profxj** [comment]
Looks good to me @finagle29 

Can you run this on at least a portion of the DevSuite
to confirm things are a-ok?   Thanks.

**Review by kbwestfall — CHANGES_REQUESTED** [review]
Great, detailed work, @finagle29 !  I had a couple of questions:

 - Do we need to add `hdul` as an attribute of `SpecObjs`? Or can avoid this by closing the hdu in the `from_fitsfile` method?
 - I wanted more detail about the change to `BitMask`, but I'm sure they're needed.  We (I can do it if you want) should propagate this to the other bit flipping methods.

File: pypeit/bitmask.py (hunk 1 above)
Comment (finagle29) [reply pypeit/bitmask.py:354]: This is necessary due to a discrepancy in the result types of bitwise operations between numpy arrays of type ``np.int32`` and ``int`` on Windows vs macOS/Linux
macOS numpy_int32
Windows numpy_intc
I'll propagate this to other bitwise operations in ``bitmask.py``

File: pypeit/specobjs.py [hunk 2]
```diff
@@ -627,6 +630,15 @@ def write_to_fits(self, subheader, outfile, overwrite=True, update_det=None,
...
             import pdb; pdb.set_trace()
+        if isinstance(_specobjs, SpecObjs):
+            # We read this in using SpecObjs.from_fitsfile()
+            # and need to clean up for Windows
+            if _specobjs.hdul is not None:
+                _specobjs.hdul.close()
+        else:
+            if self.hdul is not None:
+                # Windows cleanup
+                self.hdul.close()
```
[the remark this answers is not in the document]
Comment (finagle29) [reply pypeit/specobjs.py:641]: Good point. I'll try closing the HDU in ``from_fitsfile``

**Comment by finagle29** [comment]
Unit tests passing
in macOS
``================ 246 passed, 681 warnings in 1053.17s (0:17:33) ================`` 
in Windows
``================= 246 passed, 68 warnings in 2844.87 seconds ==================``

**Comment by finagle29** [comment]
```
--- PYPEIT DEVELOPMENT SUITE FAILED 17/66 TESTS (Masters ignored) ---
Failed tests:
    pypeit_ql: shane_kast_blue/600_4310_d55
    pypeit_ql: keck_nires/NIRES
    run_pypeit: keck_mosfire/J_multi
    run_pypeit: keck_lris_blue/multi_600_4000_d560
    run_pypeit: keck_lris_red/multi_1200_9000_d680
    pypeit_sensfunc: gemini_gnirs/32_SB_SXD
    pypeit_flux_spec: gemini_gnirs/32_SB_SXD
    pypeit_coadd_1dspec: gemini_gnirs/32_SB_SXD
    pypeit_tellfit: gemini_gnirs/32_SB_SXD
    run_pypeit: gemini_gmos/GS_HAM_R400_860
    run_pypeit: lbt_luci/LUCI-II
    run_pypeit: keck_deimos/600ZD_M_6500
    run_pypeit: keck_deimos/1200G_M_7750
    run_pypeit: keck_deimos/830G_LVM_8400
    run_pypeit: keck_deimos/830G_M_8100_26
    run_pypeit: keck_deimos/830G_M_8500
    run_pypeit: keck_deimos/830G_L_8100
```
run_pypeit: lbt_luci/LUCI-II failed with ``pypeit.pypmsgs.PypeItError: Infinities in action matrix.  B-spline fit faults.``

the gemini_gnirs/32_SB_SXD tests failed because I didn't have the Mauna Kea TellFit file installed

run_pypeit: keck_mosfire/J_multi failed with ``pypeit.pypmsgs.PypeItError: Input BITMASK keys differ from current data model!``

pypeit_ql: keck_nires/NIRES failed with ``pypeit.pypmsgs.PypeItError: You need to set an Environmental variable NIRES_MASTERS that points at the Master Calibs``

and everything else failed with ``pypeit.pypmsgs.PypeItError: Current version of Spec2DObj object in code (v1.0.2) does not match version used to write your HDU(s)!``

These were using the master branch of the dev-suite. I think the datamodel errors could plausibly be related to not being on dev-suite/develop. I'm happy to rerun on dev-suite/develop if desired.

**Comment by profxj** [comment]
All thumbs up from me.  
I encourage @kbwestfall to do the same.

**Review by kbwestfall — APPROVED** [review]
Hi @finagle29. Thanks for all of this!  Let's keep in mind the devsuite failures, in case we run into them in other PRs.

**Comment by finagle29** [comment]
Merging, but I'll keep the branch alive for now so we can reproduce the dev suite failures.

**Pull request merged**